### DISCUSSION TITLE
Redirection을 통한 JWT Token 발급 테스트

### DISCUSSION THREAD

**Discussion by sjiwon** [commerce-api-naver/commerce-solutions#68] [묻고 답하기]
Redirection을 통해서 JWT Token이 적용된 '솔루션 사용연결 URL' 로 들어온다고 파악이 되었고 이와 관련해서 RS256 Public Key 기반 복호화 테스트를 위해서 redirection flow를 진행할 수 있는 버튼? 이 있을까요?? 아니면 이에 대한 테스트는 불가능한건가요?

샘플로 제공된 토큰은 signature가 달라서 복호화가 불가능한거 같습니다

---

> JWT signature does not match locally computed signature. JWT validity cannot be asserted and should not be trusted.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

> [!IMPORTANT]
> 게시해주신 내용은 커머스솔루션 개발사 연동 관련 문의로 판단되어 관리자가 게시 공간을 이동하였습니다.
> - 최초 게시된 공간: 커머스API 내 스토어 애플리케이션 기술지원 공간 (`/commerce-api`)
> - 이동된 공간: 커머스솔루션 개발사 기술지원 공간 (`/commerce-solutions`)

JWT 토큰은 대상 솔루션 구독 상태에서 [[스마트스토어센터 > 커머스솔루션 > 솔루션 목록]](https://sell.smartstore.naver.com/#/commerce-solution/solution-list) 페이지 내 솔루션 사용 버튼을 누를 경우 지정된 솔루션 웹 사이트 URL로 랜딩되며 URL 파라미터 `token` 으로 전달됩니다.

개발중이신 솔루션 정보로 제공되는 공개키를 이용하시어 전달되는 `token` 값을 검증해보시기 바랍니다.

### 심사 승인 이전 솔루션의 구독/연동 테스트
심사 승인(심사 완료) 이전 솔루션은 대상 솔루션을 생성한 커머스ID로 로그인한 경우에만 구독/연동 테스트가 가능합니다.
솔루션을 생성한 커머스ID로 로그인 된 상태에서 [커머스솔루션마켓](https://solution.smartstore.naver.com)에 접속하실 경우 개발중인 솔루션이 표시됩니다. 이 상태에서 개발중인 솔루션의 구독 후 테스트를 진행해보시기 바랍니다.

**Comment by sjiwon** [comment]
@commerce-api-naver 혹시 '테스트 등록' 상태로 등록된 솔루션 2개가 있긴한데 동일 계정으로 솔루션 목록에 들어갔는데 표시가 안되는거는 문제가 있는건가용??

**Reply by sjiwon** [reply]
Solution ID: SOL_DOsjj4sngMQ5RuPB73L7R

**Reply by sjiwon** [reply]
심사요청 누른 후 테스트하기로 가서 앱 추가 가능한걸로 확인되었습니다 🙏

**Reply by commerce-api-naver** [reply]
@sjiwon 님, 커머스API 이민석입니다.

심사요청 이전에도 개발중인 솔루션의 추가가 가능합니다.
[[커머스API센터 > 내 솔루션]](https://apicenter.commerce.naver.com/ko/member/my-solutions/list)에 표시되는 솔루션 그리드 중 테스트가 필요한 솔루션의 "노출" 링크를 선택할 경우 대상 솔루션의 커머스솔루션마켓 상세 페이지로 이동합니다.
솔루션 관리 권한을 가진 '커머스ID'로 로그인 한 상태에서 개발중인 대상 솔루션의 커머스솔루션마켓 상세 페이지에서 솔루션 추가가 가능한 점 참고해주시기 바랍니다.
